app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,56 @@
+class MoviesController < ApplicationController
+
+  def index
+    @movies = Movie.all.sample(4)
+    @most_recent = Movie.order("created_at DESC").limit(5)
+    @mr_c_movie = Movie.find_by(title: "The Princess Diaries")
+  end
+
+  def show
+    @movie = Movie.find(params[:id])
```

Comment:
In the view this action renders, you are calling `@movie.reviews` and `@movie.comments` which makes two SQL queries from your view. It would be better to also retrieve that data here in the controller with one SQL query, that way your view is not making SQL queries (and also so that you are not making more SQL queries than you need to). Check this out:

```ruby
# this will make one query that retrieves the movie *and* all of its reviews *and* all of its comments
@movie = Movie.includes(:reviews, :comments).find(params[:id])
```
Now, when @movie.reviews and @movie.comments are called in your view, there will be no need to make any additional SQL queries, since the reviews and comments are already loaded into `@movie`.

This is called [eager loading](http://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations), and is a best practice for optimizing database queries (and thus response speed and view loading speed).